pkg/security/ebpf/c/discarders.h

```diff
@@ -39,62 +39,143 @@ int __attribute__((always_inline)) bump_discarder_revision(u32 mount_id) {
     return *revision;
 }
 
+struct discarder_state_t {
+    u64 expire_at;
+    u32 is_retained;
+};
+
+struct discarder_parameters_t {
+    u64 event_mask;
+    u64 timestamps[EVENT_MAX_ROUNDED_UP];
+    struct discarder_state_t state;
+    u32 padding;
+};
+
+void __attribute__((always_inline)) discard(struct bpf_map_def *discarder_map, void *key, u64 event_type, u64 timeout) {
+    u64 timestamp = timeout ? bpf_ktime_get_ns() + timeout : 0;
+
+    struct discarder_parameters_t *params = bpf_map_lookup_elem(discarder_map, key);
+    if (params) {
+        params->event_mask |= event_type;
+        params->timestamps[(event_type-1)&(EVENT_MAX_ROUNDED_UP-1)] = timestamp;
+    } else {
+        struct discarder_parameters_t new_params = {
+            .event_mask = event_type,
+        };
+
+        void *dst = new_params.timestamps + ((event_type-1)&(EVENT_MAX_ROUNDED_UP-1));
```

Comment:
Adding `| 0` might fix the verifier bug on Centos. I don't know if the verifier is clever enough to understand what would happen if `event_type == 0`.